db/db.go

```diff
@@ -164,7 +164,9 @@ func (db *DB) GetVersion(name string, version api.SecretVersion, from []string)
 func (db *DB) Put(name string, value []byte, from []string) (api.SecretVersion, error) {
 	db.mu.Lock()
 	defer db.mu.Unlock()
-	if !db.checkACLLocked(from, name, acl.ActionPut) {
+	if name == "" {
+		return 0, errors.New("empty secret name")
+	} else if !db.checkACLLocked(from, name, acl.ActionPut) {
```

Comment:
Slight stylistic preference: for terminal branches that return errors, I tend to prefer several standalone `if`s rather than an else-if chain, e.g.:

```
if name == "" {
  return ...
}
if !db.checkACLLocked(...) {
  return ...
}
```

Not a strong preference though, this is perfectly readable also.

Reply:
Done! I don't have a strong preference either.